DefunctRouteHandler/LeaveRedirectDefunctRouteHandler.php

```diff
@@ -36,12 +36,21 @@ public function handleDefunctRoutes(UriContextCollection $uriContextCollection)
         $referringAutoRouteCollection = $this->adapter->getReferringAutoRoutes($uriContextCollection->getSubjectObject());
 
         foreach ($referringAutoRouteCollection as $referringAutoRoute) {
-            if (false === $uriContextCollection->containsAutoRoute($referringAutoRoute)) {
-                $newRoute = $uriContextCollection->getAutoRouteByTag($referringAutoRoute->getAutoRouteTag());
+            if (true === $uriContextCollection->containsAutoRoute($referringAutoRoute)) {
+                continue;
+            }
+
+            $newRoute = $uriContextCollection->getAutoRouteByTag($referringAutoRoute->getAutoRouteTag());
 
-                $this->adapter->createRedirectRoute($referringAutoRoute, $newRoute);
+            if (!$newRoute) {
```

Comment:
Throwing an exception is probably not the thing to do.. it could happen that a locale is removed from the subject document, so there could be extant referring routes in the old locales.

Skipping is probably the best option, although it could hide bugs.

We should probably remove the "new route" logic all together, as we should reference the content not the route.